panic: runtime error: on swarm:latest

Greetings,

Im getting lots of this today, on a 1master, 2nodes.  any ideas on this?

master
Distributor ID:	Debian
Description:	Debian GNU/Linux 7.8 (wheezy)
Release:	7.8
Codename:	wheezy

node1
Distributor ID:	Debian
Description:	Debian GNU/Linux 7.8 (wheezy)
Release:	7.8
Codename:	wheezy

node2
CoreOS stable (557.2.0)

all running
go version go1.4.1 linux/amd64

```Go
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x48a9dd]

goroutine 16 [running]:
runtime.panic(0x7aaf20, 0xa09713)
	/usr/src/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/docker/swarm/discovery/token.(*TokenDiscoveryService).Register(0xc208032d80, 0xc20804e720, 0x11, 0x0, 0x0)
	/go/src/github.com/docker/swarm/discovery/token/token.go:94 +0x2cd
main.join(0xc208052240)
	/go/src/github.com/docker/swarm/join.go:42 +0x695
github.com/codegangsta/cli.Command.Run(0x800030, 0x4, 0x800010, 0x1, 0x839c30, 0x15, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/swarm/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:113 +0xeb2
github.com/codegangsta/cli.(*App).Run(0xc2080015f0, 0xc20800e000, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/docker/swarm/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:156 +0xaea
main.main()
	/go/src/github.com/docker/swarm/main.go:121 +0xaa3

goroutine 19 [finalizer wait, 196 minutes]:
runtime.park(0x4177a0, 0xa0e248, 0xa0c489)
	/usr/src/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0xa0e248, 0xa0c489)
	/usr/src/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
	/usr/src/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
	/usr/src/go/src/pkg/runtime/proc.c:1445

goroutine 23 [select]:
net.func·019()
	/usr/src/go/src/pkg/net/dnsclient_unix.go:184 +0x2ab
created by net.loadConfig
	/usr/src/go/src/pkg/net/dnsclient_unix.go:212 +0x153
```

--- a/discovery/token/token.go
+++ b/discovery/token/token.go
@@ -49,9 +49,7 @@ func (s *TokenDiscoveryService) Fetch() ([]*discovery.Entry, error) {
 		return nil, err
 	}
 
-	if resp.Body != nil {
-		defer resp.Body.Close()
-	}
+	defer resp.Body.Close()
 
 	var addrs []string
 	if resp.StatusCode == http.StatusOK {
@@ -81,9 +79,12 @@ func (s *TokenDiscoveryService) Register(addr string) error {
 	resp, err := http.Post(fmt.Sprintf("%s/%s/%s", s.url,
 		"clusters", s.token), "application/json", buf)
 
-	// Force connection close
+	if err != nil {
+		return err
+	}
+
 	resp.Body.Close()
-	return err
+	return nil
 }
 
 // CreateCluster returns a unique cluster token

